### PR TITLE
fix(offsite): comprehensive operator-experience logging review (LLMO-6973)

### PR DESCRIPTION
**Pull request by grigoreadobe**
## Summary

Introduces OUTCOME.DEGRADED as warn()'s new default outcome (was FAILURE) — the root cause behind dozens of non-fatal, self-healed conditions reading as hard failures, and behind the same skip scenario resolving to a different outcome depending on which content-type handler ran. Makes every warn() call's outcome explicit going forward.

Backfills reason/reasonCategory across every failure/warn call that was missing one; surfaces three Semrush diagnostic reasons that had drifted out of the Splunk-visible log fields; adds a distinct PEER.GUIDELINE_STORE so the sentiment/topic store is no longer indistinguishable from the URL store; migrates every hand-rolled logging bypass (store-client.js, offsite-brand-presence-enrichment.js, offsite-brand-presence-postgrest.js, semrush-entitlement.js) onto the shared createOffsiteLogger, closing the one place raw error text was interpolated into a log message.

Adds writeAction/snapshotAction/dispatchKind fields to disambiguate merged events without renaming or splitting any of them. Instruments three previously-silent transitions (empty URL-store/guideline-store results, "nothing eligible" retention runs, youtube's uninstrumented Slack send). Standardizes message wording and field names (companyName, messageType) that had diverged across the four near-identical content-type handlers. Removes dead guideline-store logging code.

No event was renamed or split.

## Change Management

```yaml
cm-assessment: v1
changeType: standard
impact: unnoticeable
risk: minor
changeApprovedBy: ["Dominique Jäggi"]
rationale: "Auto-added baseline (no PR assessment present at CI time); refine via the cmr skill if this change is higher-impact."
```

### CODE REVIEW

File: src/utils/store-client.js [hunk 1]
```diff
@@ -286,12 +280,15 @@ export default class StoreClient {
     const guidelines = guidelineItems.map(toSentimentGuidelineJson);
 
     if (guidelines.length === 0) {
+      olog.skip('audit_orchestration_brand_guidelines_resolved', 'No guidelines or topics found in URL Store', {
```
Comment (MysticatBot) [inline src/utils/store-client.js]: issue (blocking): The message reads "No guidelines or topics found in URL Store" but the `peer` field is `PEER.GUIDELINE_STORE`. Operators searching Splunk by the peer they see (`guideline_store`) will not find this line by text, and vice versa. Update the message to say "No guidelines or topics found in Guideline Store" (or remove the store name from the message and let the peer field speak for itself).

File: src/utils/offsite-brand-presence-postgrest.js [hunk 2]
```diff
@@ -186,32 +182,34 @@ export async function loadBrandPresenceDataFromPostgrest({
...
-        event: 'data_acquisition_bp_data_postgres_read', outcome: OUTCOME.SKIP, peer: PEER.POSTGRES, direction: 'inbound', count: 0, reason: 'no_executions', reasonCategory: 'config', siteId,
-      }));
+      olog.skip('data_acquisition_bp_data_postgres_read', 'No execution rows found', {
+        peer: PEER.POSTGRES, direction: 'inbound', count: 0, reason: 'no_executions', reasonCategory: 'config', siteId,
+      });
       return null;
     }
 
     const rows = mapExecutionsToLegacyBrandPresenceRows(executions);
     if (rows.length === 0) {
```
Comment (MysticatBot) [inline src/utils/offsite-brand-presence-postgrest.js:197]: issue (blocking): This catch was promoted from `log?.warn(...)` to `olog.failure(...)` (error level, outcome=failure). However the system recovers gracefully here - the caller falls through to the SharePoint path. Logging a terminal `failure` outcome for a condition the service handles via fallback will trigger error-level Splunk alerts and potential pages for a non-terminal event.

Fix: Keep this at `olog.warn(...)` with `outcome: OUTCOME.DEGRADED` (matching the PR thesis that non-fatal recovered conditions should be DEGRADED, not FAILURE), or document in the on-call runbook that this specific event+outcome is non-paging.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @grigoreadobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - two operator-facing issues need fixing before merge.
**Complexity: HIGH** - large diff (29 files); logging taxonomy change across all offsite handlers.
_Changes: Introduces OUTCOME.DEGRADED as warn()'s new default, backfills reason/reasonCategory across all warn/failure calls, adds PEER.GUIDELINE_STORE, migrates hand-rolled logging onto createOffsiteLogger, removes dead guideline-store code (29 files)._

## Must fix before merge

1. [Important] Log message says "URL Store" but peer is `guideline_store` - `src/utils/store-client.js:283` (details inline)
2. [Important] `.failure()` emits at error level for a recoverable fallback path - `src/utils/offsite-brand-presence-postgrest.js:196` (details inline)

<details>
<summary>Non-blocking (5): minor issues and suggestions</summary>

- nit: `store-client.js:203,259` uses string literal `'store-client'` as audit slug instead of the `AUDIT` enum - consider `AUDIT.STORE_CLIENT` for discoverability
- nit: `youtube-analysis/guidance-handler.js:283` new Slack catch omits `direction: 'outbound'`, unlike all other peer-interaction log sites
- suggestion: `offsite-brand-presence-postgrest.js:76` - `olog.warn()` with `outcome: OUTCOME.SUCCESS` is a confusing Splunk query pattern; consider `olog.success()` with a caveat field
- nit: `writeAction: 'suggestions_synced'` on failure log paths reads like past-tense success; `'suggestions_sync'` would better convey the attempted action (affects 4 guidance-handler files)
- nit: `common/offsite-retention.js:95` new debug log for zero-eligible path omits `peer`/`direction` unlike the success path of the same event

</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 9s | <b>Cost:</b> $5.03 | <b>Commit:</b> <code>aff27d280053c0a9c6f88d9d271bb6d30d65f598</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @grigoreadobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - both prior blocking findings resolved; no new blocking issues.
**Complexity: HIGH** - large diff (34 files); logging taxonomy change across all offsite handlers.
_Changes: Introduces OUTCOME.DEGRADED as warn()'s new default, backfills reason/reasonCategory across all warn/failure calls, adds PEER.GUIDELINE_STORE, migrates hand-rolled logging onto createOffsiteLogger, removes dead guideline-store code (34 files)._

<details>
<summary>Non-blocking (4): minor issues and suggestions</summary>

- suggestion: `resolveForwardedUrlLimit` still passes raw platform `log` to `resolveMystiqueUrlLimit` (which expects a bound `olog`) - complete the migration to `(auditContext, olog)` in a follow-up to close the last unmigrated sibling - `src/utils/offsite-audit-utils.js`
- nit: `youtube-analysis/guidance-handler.js:284` new Slack catch omits `direction: 'outbound'`, unlike all other peer-interaction log sites
- nit: `writeAction: 'suggestions_synced'` on failure log paths reads like past-tense success; `'suggestions_sync'` would better convey the attempted action (affects 4 guidance-handler files)
- nit: `store-client.js:206,262` uses string literal `'store-client'` as audit slug instead of an `AUDIT` enum member - Splunk queries filtering on known audit values will miss these lines

</details>

## Previously flagged, now resolved

- Log message/peer mismatch in store-client.js ("URL Store" vs guideline_store) - message now correctly says "Guideline Store"
- `.failure()` at error level for recoverable PostgREST fallback path - changed to `.warn()` with outcome=degraded

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 35s | <b>Cost:</b> $7.90 | <b>Commit:</b> <code>ae0e0752cd638e8b3df13b1df12a420aa4bfbab9</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @grigoreadobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - both prior blocking findings resolved; incremental taxonomy refinements are consistent and correct.
**Complexity: HIGH** - large diff (37 files); logging taxonomy change across all offsite handlers.
_Changes: Continues the outcome/level taxonomy audit - demotes recoverable failures to warn+degraded, promotes truly unrecoverable infra gaps to failure, backfills outcome fields, and standardizes level semantics across all offsite handlers (37 files)._

<details>
<summary>Non-blocking (5): minor issues and suggestions</summary>

- suggestion: `no_company_name` across all 4 content-type handlers is changed from OUTCOME.SKIP to OUTCOME.DEGRADED, but the audit returns `success: false` and produces nothing - DEGRADED implies the system continued with partial output, while this is a clean abort. SKIP (precondition unmet, no-op) fits the semantics better. The `no_competitors` change to DEGRADED is correct (audit proceeds with reduced Share of Voice).
- nit: `youtube-analysis/guidance-handler.js:287` new Slack catch omits `direction: 'outbound'`, unlike all other peer-interaction log sites
- nit: `writeAction: 'suggestions_synced'` on failure log paths reads like past-tense success; `'suggestions_sync'` would better convey the attempted action (affects 4 guidance-handler files)
- nit: `store-client.js:206,262` uses string literal `'store-client'` as audit slug instead of the `AUDIT` enum - Splunk queries filtering on known audit values will miss these lines
- suggestion: Consider adding an ADR (`docs/decisions/00X-offsite-outcome-degraded.md`) documenting the DEGRADED introduction and the warn() default change - the code comments have excellent rationale that would benefit from formal ADR status per repo conventions

</details>

## Previously flagged, now resolved

- Log message/peer mismatch in store-client.js ("URL Store" vs guideline_store) - resolved in prior commit
- `.failure()` at error level for recoverable PostgREST fallback path - resolved in prior commit


---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 56s | <b>Cost:</b> $7.43 | <b>Commit:</b> <code>cf5bde6f3627062037343fd96610377b0a93cd7a</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @grigoreadobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - two issues need resolution before merge.
**Complexity: HIGH** - large diff (39 files); logging taxonomy change across all offsite handlers.
_Changes: Drops `reasonCategory` from all logging calls, finalizes outcome/level taxonomy, migrates last hand-rolled logging onto `createOffsiteLogger`, softens snapshot failure handling to avoid aborting primary persistence (39 files)._

## Must fix before merge

1. [Important] Comment/code contradiction: comment says "must not page, outcome=degraded reflects that" but code uses `.failure()` with `OUTCOME.FAILURE` - `src/offsite-brand-presence/handler.js:1051` (details inline)
2. [Important] ADR 003 is now stale: documents `reasonCategory` as essential for alerting ("the primary alerting split going forward") but the PR removes it from all code without updating the ADR - `docs/decisions/003-offsite-event-taxonomy-phase-boundaries.md`

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- nit: `store-client.js` uses hardcoded string `'store-client'` as audit slug rather than an `AUDIT` enum member - Splunk queries on known audit values will miss these lines
- nit: `offsite-retention.js:89` new warn-level log for the zero-eligible snapshots path will fire on every healthy site every refresh cycle - consider demoting to skip/info once operators confirm they trust the distinction

</details>

## Previously flagged, now resolved

- Log message/peer mismatch in store-client.js ("URL Store" vs guideline_store) - resolved in prior commit
- `.failure()` at error level for recoverable PostgREST fallback path - resolved in prior commit

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 3m 52s | <b>Cost:</b> $9.47 | <b>Commit:</b> <code>b6a3d02fdb48fb5c38ecf4b4353dfcd56e147f35</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @grigoreadobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - both prior blocking findings resolved; incremental commit is clean and well-tested.
**Complexity: HIGH** - large diff (41 files); logging taxonomy change across all offsite handlers.
_Changes: Addresses prior review findings - updates stale ADR 003, adds ADR 005 documenting reasonCategory removal, fixes comment/code contradiction in handler.js, replaces hardcoded audit slug with toLogAudit() enum mapping (41 files)._
Note: Recommend a human read before merge - this change amends architectural decision records in `docs/decisions/`. The bot review is a complement to, not a replacement for, a human read here.

<details>
<summary>Non-blocking (4): minor issues and suggestions</summary>

- suggestion: `toLogAudit` default branch passes the raw `auditType` string through, which silently emits a non-enum value to Splunk if a new audit type is added without updating the switch. Consider a follow-up PR adding a schema test that enumerates `Audit.AUDIT_TYPES` and asserts each is mapped - `src/utils/store-client.js:84`
- suggestion: Add a test case passing a non-null unknown string (e.g. `'some-future-audit'`) to verify the passthrough behavior documented in the JSDoc, since the current default-branch coverage comes only from the `undefined` path - `test/utils/store-client.test.js`
- nit: `writeAction: 'write_failed'` in `offsite-refresh.js:144` conflates the action being attempted with its result; the failure is already expressed by `reason: 'opportunity_write_failed'` + `outcome=failure`. Consider `writeAction: 'refresh_attempted'` for consistency with the other sites
- suggestion: Confirm that existing Splunk alerts cover `outcome=degraded` on snapshot operations (`audit_persistence_snapshot_opportunity_read`, `audit_persistence_snapshot_opportunity_write`). The behavioral change from throw-to-return-null in `findSnapshotByTriggerAuditId` means persistent DB failures now surface only at warn/degraded level, not error/failure

</details>

## Previously flagged, now resolved

- Comment/code contradiction in handler.js:1051 (said "must not page" but used .failure()) - comment rewritten to explain why orphaned DRS jobs warrant paging
- ADR 003 stale (documented reasonCategory as essential while PR removed it) - ADR 003 marked Superseded, new ADR 005 documents the removal decision
- store-client.js hardcoded 'store-client' audit slug - replaced with toLogAudit() mapping to proper AUDIT enum members, with comprehensive tests

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 29s | <b>Cost:</b> $7.53 | <b>Commit:</b> <code>b013051b4e5658549a24c3852658c3ce9037742c</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
